### PR TITLE
CSSTUDIO-2002: handle app errors and unknown routes

### PR DESCRIPTION
**Pull request by cjenkscybercom**
## Summary of Changes

- Adds error boundary page (for generic unhandled runtime app errors)
Screenshot 2024-03-13 at 13 59 57

- Adds 404 page handling for unknown routes
Screenshot 2024-03-13 at 14 00 13

## Visual Inspection

- [ ] Conformance to Markdown styles: https://olog.esss.lu.se/Olog/help/CommonmarkCheatsheet
    - [ ] ...when viewing a log entry
    - [ ] ...when previewing HTML while writing a description
    - [ ] ...when viewing a log entry in the group view
- [ ] Scroll is possible (elements don't overflow their container, and they are scrollable)
    - [ ] ...search result list
    - [ ] ...log entry group view list
    - [ ] ...log entry single view
    - [ ] ...create new log entry page
- [ ] Overall layout fills full width and height of viewport
- [ ] Pagination element doesn't overflow into other elements

### CODE REVIEW

**Comment by cjenkscybercom** [comment]
Note: Please ignore (flaky) test failures; they all pass locally, I've created a backlog item to investigate as it seems to be specifically a GitHub problem with the cutbacks to performance of GH action runner resources and how Cypress is sensitive to that.

File: src/providers/RouterProvider.js [hunk 1]
```diff
@@ -40,12 +43,21 @@ export const routes = [
             {
                 path: "logs/:id/history",
                 element: <LogHistoryView />
+            },
+            {
+                path: "error-test",
```
Comment (jacomago) [inline src/providers/RouterProvider.js:48]: Is it possible to only have this in a dev profile/build somehow?

**Review by georgweiss — COMMENTED** [review]
Not entirely happy with the graphics in "...Whoops!..." 
Propose to keep only text.

**Comment by cjenkscybercom** [comment]
> Not entirely happy with the graphics in "...Whoops!..." Propose to keep only text.

"But it's funnnn!"

Fair enough, can remove it; just fire emoji unicode, might look funny in some ancient browsers 🤷

**Pull request merged**